JavaScript/JavaScriptSDK/Sender.ts

```diff
@@ -168,22 +180,17 @@ module Microsoft.ApplicationInsights {
                 // Send data only if disableTelemetry is false
                 if (!this._config.disableTelemetry()) {
 
-                    if (this._buffer.length) {
-                        // compose an array of payloads
-                        var batch = this._config.emitLineDelimitedJson() ?
-                            this._buffer.join("\n") :
-                            "[" + this._buffer.join(",") + "]";
-
+                    if (this._buffer.count() > 0) {
                         // invoke send
-                        this._sender(batch, isAsync, this._buffer.length);
+                        this._sender(this._buffer.getItems(), isAsync, this._buffer.count());
```

Comment:
This looks confusing - you pass items in but sender actually calls .batchPayloads itself and sends it as. My advice is that you either pass the whole buffer and let _sender operate with it or pass everything as arguments.